src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterInstructions } from './types/FilterInstructions';
+import { TodosType } from './types/TodosType';
+import { TodoModal } from './components/TodoModal';
+
+const getPreparedTodos = (
+  todosList: Todo[],
+  filterInstruction: FilterInstructions,
+) => {
+  return todosList.filter(todo => {
+    const containsSubstringInName = todo.title
+      .toLowerCase()
+      .includes(filterInstruction.todoName.toLowerCase());
+
+    if (filterInstruction.todosType === TodosType.Active) {
+      return !todo.completed && containsSubstringInName;
+    }
+
+    if (filterInstruction.todosType === TodosType.Completed) {
+      return todo.completed && containsSubstringInName;
+    }
+
+    return containsSubstringInName;
+  });
+};
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [chosenTodo, setchosenTodo] = useState<Todo | null>(null);
+
+  const [filterInstructions, setFilterInstructions] =
+    useState<FilterInstructions>({
+      todosType: TodosType.All,
+      todoName: '',
+    });
+
+  useEffect(() => {
+    getTodos().then(setTodoList);
+  }, []);
+
+  const handleChoosingTodo = (todo: Todo | null) => {
```

Comment:
Please, consider renaming your methods following the naming conventions for methods. For example, instead of 'handleChoosingTodo', you can rename it to 'onTodoSelect'. Here you can find more details about naming conventions: https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364